openwisp_network_topology/integrations/device/base/models.py

```diff
@@ -69,21 +90,28 @@ def auto_create_openvpn(cls, node):
         if not device:
             return
 
-        device_node = cls(device=device, node=node)
-        try:
-            device_node.full_clean()
-            device_node.save()
-            # Update organization of the node. This is required
-            # when topology is shared.
-            if node.organization_id is None:
-                node.organization_id = device.organization_id
-                node.save(update_fields=['organization_id'])
-        except Exception:
-            logger.exception('Exception raised during auto_create_openvpn')
+        return cls.save_device_node(device, node)
+
+    @classmethod
+    def auto_create_wireguard(cls, node):
+        allowed_ips = node.properties.get('allowedIps')
+        if not allowed_ips:
             return
-        else:
-            logger.info(f'DeviceNode relation created for {node.label} - {device.name}')
-            return device_node
+        Device = load_model('config', 'Device')
+        ip_addresses = [str(IPv4Network(cidr).network_address) for cidr in allowed_ips]
+        device_filter = models.Q(config__vpnclient__ip__ip_address__in=ip_addresses)
```

Comment:
I used allowed Ips in this case and modified query accordingly.

Reply:
the ip address is going to be only the ip address, while the allowed ip of wireguard is CIDR notation